### ISSUE TITLE
CFE_ES_GetPoolBuf() return value is confusing

### ISSUE THREAD

**Issue by CDKnightNASA** [nasa/cFE#2136]
On error, CFE_ES_GetPoolBuf() returns a status code, but on success, it returns the size allocated (which is actually the Size input parameter, so the return value is redundant.) I suggest either make the Size parameter an in/out parameter, or just return a CFE status and the caller can assume CFE_SUCCESS means the memory size requested was allocated.

https://github.com/nasa/cFE/blob/318f3451304e0d7e2269054fe170af99189ec560/modules/es/fsw/src/cfe_es_mempool.c#L424

**Comment by skliper** [comment]
Agreed, although in theory API change requires a "major update" unless we can do it by just adding an API and deprecating this one. If we do it here we might want to tackle the rest of the "multiple success return codes" issue:
- #483
